New sdk maintainers rules (#665)
cloudevents/spec: governance/SDK.md

## New maintainers

> Maintainers are a crucial part of this community: they keep alive our projects

Reviewer: ```suggestion
Maintainers are a crucial part of this community. They keep our projects alive
```

and ensure both development and maintenance. We define a set of criteria,
followed by a vote, to include an active contributor as maintainer of one or

> more sdk projects. The active contributor may propose himself to become a

Reviewer: ```suggestion
more SDK projects. An active contributor may propose themselves as a
```

> maintainer, although we encourage the community to include him as part of the
> group.

Reviewer: ```suggestion
maintainer, although we encourage the existing maintainers to nominate active contributors proactively.
```

> The criteria for a contributor to `sdk-x` to become part of `sdk-x-maintainers`

Reviewer: ```suggestion
The criteria for a contributor to become a maintainer for a given SDK repository
```

is:

1. At least 20 reviews on eligible PRs

> 1. At least 20 submitted eligible PRs as author

Reviewer: This is a pretty high bar for some repositories. Even as active as sdk-javascript is, there have only been 20 PRs in the last month, and it would be unusual for a community member to contribute anywhere near that much. I might suggest something looser such as.

> 1. Regular reviews of eligible pull requests over a period of 30 days
> 1. Regular contributions to the project in the form of eligible pull requests over a period of 30 days

Author: I disagree for a couple of reasons:

* I think the criteria should be well defined, number based. This makes sure rules are always followed and there are no shady situations. I wanna make sure nobody can say "this guy had a different treatment from other people"
* 20 both for PRs and reviews seems pretty reasonable. We're not defining how you become a contributor, but how you become a maintainer and you get full powers in the project like everybody else. If the project is active, 20 PRs and 20 reviews is something you get in ~2 months, which IMO is a good time frame to go from 0 to maintainer.

Reviewer: My experience with the Node.js project was quite different than this. I consider that project to be very well managed, and their requirements for becoming a maintainer are definitely looser than this, while at the same time consideration is given to a lot of activities outside of actual PRs.

https://nodejs.org/en/get-involved/contribute/#becoming-a-collaborator

* code commits and pull requests
* documentation commits and pull requests
* comments on issues and pull requests
* contributions to the Node.js website
* assistance provided to end users and novice contributors
* participation in Working Groups
* other participation in the wider Node.js community

And they allow for contributors to bring up themselves for consideration if they feel the requirements have been met.

> If individuals making valuable contributions do not believe they have been considered for commit access, they may log an issue or contact a TSC member directly.

> When we refer to the term _eligible PR_, we explicit PRs with substantial

Reviewer: ```suggestion
When we refer to the term _eligible pull requests_, we explicitly mean pull requests with substantial
```

contributions to the project. Because a _"substantial contribution"_ may vary
between the different technologies used, each project defines in the
`CONTRIBUTING.md` (available at the root of the Github Repository) what

> _eligible PR_ means for it.

Reviewer: In general, I am not sure how comfortable I am with this. Documentation changes, adding test cases and other minor contributions are certainly valuable, and I would merit consistency, care and engagement, just as important as big sweeping pull requests.

Author: Sure, that's why I say "eligible PR" is something defined by each sdk. A typo in the doc for me it's not valuable, a bug fix with a new test case for me it's more than valuable. I don't say how this criteria should be defined here, because it's up to the single projects

Reviewer: I see what you are saying. But this puts the onus on each SDK to define up front what constitutes grounds for an eligible PR, and personally, I don't feel comfortable doing that for sdk-javascript, given that I also think consideration should be given to other, "softer" contributions as I mention in my comment above.

> If the contributor miss one of those criteria, he cannot ask to become

Reviewer: ```suggestion
If a contributor does not meet these criteria, they should not be considered for approval as a
```

maintainer of the project, **unless** the situation defined in
[handover the project](#handover-to-a-new-maintainergroup-of-maintainers)
applies.

> As soon as the contributor meets the criteria, he can ask (or an existing
> maintainer can ask) to proceed with a vote using the

Reviewer: ```suggestion
Once a contributor has met the above criteria, they, or an existing maintainer,
can ask to proceed with a vote using the
```
